### PR TITLE
Add basic test for the sleep filter

### LINKED ISSUE #1: No tests / Tests missing
Hi,
  this plugin has no test at all, this is a situation that should be amended soon, so for example changes in both the plugin or the logstash core himself, are properly tested in the CI environment.

If you are giving this plugin some test, feel free to update https://github.com/elastic/logstash/issues/3740 so we keep track of it.

### PR DESCRIPTION
**Pull request by purbon**
Fixes #1

### CODE REVIEW

**Comment by ph** [comment]
On it thanks @pere

File: spec/filters/sleep_spec.rb [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+      it "should sleep for N seconds and continue" do
+        expect(subject).to receive(:sleep).with(5).exactly(4).times
+        messages.times do
+          subject.filter(event)
+        end
+      end
+
+    end
+  end
+
```
Comment (ph) [inline spec/filters/sleep_spec.rb]: Can we remove a bit of space before the end?

File: spec/filters/sleep_spec.rb [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+require_relative "../spec_helper"
+require "logstash/plugin"
+require "logstash/event"
 
 describe LogStash::Filters::Sleep do
+
+  let(:time) { 1 }
+  subject { LogStash::Filters::Sleep.new("time" => time) }
+
+  let(:properties) { {:name => "foo" } }
```
Comment (ph) [inline spec/filters/sleep_spec.rb]: add a space before `:name` since we have one after "foo"

**Comment by ph** [comment]
I think we don't have a test case when using the `replay` option? But it shouldnt block merging this.

Made small minor comments, Code LGTM, Test passed.

**Comment by purbon** [comment]
@ph wanted to add initial test, I agree doing reply is another nice test to add.

**Pull request closed without merging**